pandas/core/dtypes/dtypes.py

```diff
@@ -403,6 +403,7 @@ def _hash_categories(categories, ordered=True):
         from pandas.core.util.hashing import (
             hash_array, _combine_hash_arrays, hash_tuples
         )
+        from pandas.core.dtypes.common import is_datetime64tz_dtype, _NS_DTYPE
```

Comment:
these can be imported at the top (the hashing ones are not for other reasons)

Reply:
That causes a circular import. dtypes.common imports from dtypes.dtypes.